### ISSUE TITLE
torchvision issue: __array__() takes 1 positional argument but 2 were given

### ISSUE THREAD

**Issue by MalteEbner** [lightly-ai/lightly#433]
The latest unittests on master fail after merging the commit `restricts filepath to 255 characters, which is a common limit (#432)`
https://github.com/lightly-ai/lightly/actions
As the error did not occur when running unittests before, I suspect it is caused by a newer version of an installed package.
The only packages updated were Pillow (from 8.2.0 to 8.3.0) and google-auth (from 1.32.0 to 1.32.1), I checked with a tool.

The error occurs when casting a torch tensor to a numpy array and specifying its dtype
```
File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/torchvision/transforms/functional_pil.py", line 96, in adjust_hue
np_h = np.array(h, dtype=np.uint8)
TypeError: __array__() takes 1 positional argument but 2 were given
```

Installed packages with ERROR:
`Jinja2-3.0.1 MarkupSafe-2.0.1 Pillow-8.3.0 PyYAML-5.4.1 Pygments-2.9.0 absl-py-0.13.0 aiohttp-3.7.4.post0 alabaster-0.7.12 antlr4-python3-runtime-4.8 appdirs-1.4.4 async-timeout-3.0.1 attrs-21.2.0 av-8.0.3 babel-2.9.1 cachetools-4.2.2 certifi-2021.5.30 cfgv-3.3.0 chardet-4.0.0 cycler-0.10.0 distlib-0.3.2 docutils-0.16 execnet-1.9.0 filelock-3.0.12 fsspec-2021.6.1 future-0.18.2 google-auth-1.32.1 google-auth-oauthlib-0.4.4 grpcio-1.38.1 hydra-core-1.1.0 identify-2.2.10 idna-2.10 imagesize-1.2.0 importlib-metadata-4.6.0 importlib-resources-5.2.0 iniconfig-1.1.1 kiwisolver-1.3.1 lightly-1.1.14 lightly-utils-0.0.1 markdown-3.3.4 matplotlib-3.4.2 multidict-5.1.0 nodeenv-1.6.0 numpy-1.21.0 oauthlib-3.1.1 omegaconf-2.1.0 opencv-python-4.5.2.54 packaging-20.9 pluggy-0.13.1 pre-commit-2.13.0 protobuf-3.17.3 py-1.10.0 pyDeprecate-0.3.0 pyasn1-0.4.8 pyasn1-modules-0.2.8 pyparsing-2.4.7 pytest-6.2.4 pytest-forked-1.3.0 pytest-xdist-2.3.0 python-dateutil-2.8.1 pytorch-lightning-1.3.7.post0 pytz-2021.1 requests-2.25.1 requests-oauthlib-1.3.0 responses-0.13.3 rsa-4.7.2 six-1.16.0 snowballstemmer-2.1.0 sphinx-4.0.2 sphinx-gallery-0.9.0 sphinx-rtd-theme-0.5.2 sphinxcontrib-applehelp-1.0.2 sphinxcontrib-devhelp-1.0.2 sphinxcontrib-htmlhelp-2.0.0 sphinxcontrib-jsmath-1.0.1 sphinxcontrib-qthelp-1.0.3 sphinxcontrib-serializinghtml-1.1.5 tensorboard-2.4.1 tensorboard-plugin-wit-1.8.0 toml-0.10.2 torch-1.9.0 torchmetrics-0.4.0 torchvision-0.10.0 tox-3.23.1 tqdm-4.61.1 typing-extensions-3.10.0.0 urllib3-1.26.6 virtualenv-20.4.7 werkzeug-2.0.1 wheel-0.36.2 yarl-1.6.3 zipp-3.4.1`
Installed packages without ERROR:
`Jinja2-3.0.1 MarkupSafe-2.0.1 Pillow-8.2.0 PyYAML-5.4.1 Pygments-2.9.0 absl-py-0.13.0 aiohttp-3.7.4.post0 alabaster-0.7.12 antlr4-python3-runtime-4.8 appdirs-1.4.4 async-timeout-3.0.1 attrs-21.2.0 av-8.0.3 babel-2.9.1 cachetools-4.2.2 certifi-2021.5.30 cfgv-3.3.0 chardet-4.0.0 cycler-0.10.0 distlib-0.3.2 docutils-0.16 execnet-1.9.0 filelock-3.0.12 fsspec-2021.6.1 future-0.18.2 google-auth-1.32.0 google-auth-oauthlib-0.4.4 grpcio-1.38.1 hydra-core-1.1.0 identify-2.2.10 idna-2.10 imagesize-1.2.0 importlib-metadata-4.6.0 importlib-resources-5.2.0 iniconfig-1.1.1 kiwisolver-1.3.1 lightly-1.1.14 lightly-utils-0.0.1 markdown-3.3.4 matplotlib-3.4.2 multidict-5.1.0 nodeenv-1.6.0 numpy-1.21.0 oauthlib-3.1.1 omegaconf-2.1.0 opencv-python-4.5.2.54 packaging-20.9 pluggy-0.13.1 pre-commit-2.13.0 protobuf-3.17.3 py-1.10.0 pyDeprecate-0.3.0 pyasn1-0.4.8 pyasn1-modules-0.2.8 pyparsing-2.4.7 pytest-6.2.4 pytest-forked-1.3.0 pytest-xdist-2.3.0 python-dateutil-2.8.1 pytorch-lightning-1.3.7.post0 pytz-2021.1 requests-2.25.1 requests-oauthlib-1.3.0 responses-0.13.3 rsa-4.7.2 six-1.16.0 snowballstemmer-2.1.0 sphinx-4.0.2 sphinx-gallery-0.9.0 sphinx-rtd-theme-0.5.2 sphinxcontrib-applehelp-1.0.2 sphinxcontrib-devhelp-1.0.2 sphinxcontrib-htmlhelp-2.0.0 sphinxcontrib-jsmath-1.0.1 sphinxcontrib-qthelp-1.0.3 sphinxcontrib-serializinghtml-1.1.5 tensorboard-2.4.1 tensorboard-plugin-wit-1.8.0 toml-0.10.2 torch-1.9.0 torchmetrics-0.4.0 torchvision-0.10.0 tox-3.23.1 tqdm-4.61.1 typing-extensions-3.10.0.0 urllib3-1.26.6 virtualenv-20.4.7 werkzeug-2.0.1 wheel-0.36.2 yarl-1.6.3 zipp-3.4.1`

**Comment by philippmwirth** [comment]
Looks like torchvision has problems with the new PIL version. I can reproduce with:

```python
import PIL
import torchvision.transforms.functional as F

print(torchvision.__version__)
print(PIL.__version__)

img = PIL.Image.open('path/to/jpg')
F.adjust_hue(img, 0.)
```

Output:
```
0.8.2+cu101
8.3.0

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/envs/boris/lib/python3.7/site-packages/torchvision/transforms/functional.py", line 712, in adjust_hue
    return F_pil.adjust_hue(img, hue_factor)
  File "/opt/conda/envs/boris/lib/python3.7/site-packages/torchvision/transforms/functional_pil.py", line 197, in adjust_hue
    np_h = np.array(h, dtype=np.uint8)
TypeError: __array__() takes 1 positional argument but 2 were given
```

@MalteEbner let's make a requirement for `Pillow<=8.2.0` until the problem is fixed.

**Comment by IgorSusmelj** [comment]
I would assume this error will be resolved with pillow 8.3.1: https://github.com/python-pillow/Pillow/issues/5571

**Comment by IgorSusmelj** [comment]
Close issue as new pillow version fixes it: 
![image](https://user-images.githubusercontent.com/1522766/124895653-06d01a00-dfdd-11eb-9f12-1d30351d9440.png)
